Python/Product/Analysis/Parsing/Tokenizer.cs

```diff
@@ -844,8 +845,15 @@ public Tokenizer(PythonLanguageVersion version, ErrorSink errorSink = null, Toke
         }
 
         public static bool IsIdentifierChar(char ch) {
-            if (IsIdentifierStartChar(ch)) {
-                return true;
+            // Latin1 case
+            if (ch <= byte.MaxValue) {
+                return ch <= 'z'
+                    ? ch <= 'Z'
+                      ? ch >= 'A' || ch >= '0' && ch <= '9'
```

Comment:
Identifiers can't start with a number.

This would be as much of a gain if it just checked ASCII letters, I suspect.

Reply:
Yep, ASCII is even better.

Code below returns `true` for `UnicodeCategory.DecimalDigitNumber`:
```cs
            switch (char.GetUnicodeCategory(ch)) {
                // Supported categories for continuing an identifier
                case UnicodeCategory.NonSpacingMark:
                case UnicodeCategory.SpacingCombiningMark:
                case UnicodeCategory.DecimalDigitNumber:
                case UnicodeCategory.ConnectorPunctuation:
                    return true;
            }
```